File: dotnet-maui-instana-integration/android/native/newbinding/src/main/java/com/example/newbinding/DotnetNewBinding.java
package com.example.newbinding;

import android.app.Activity;
import android.util.Log;

import com.instana.android.CustomEvent;
import com.instana.android.Instana;
import com.instana.android.core.HybridAgentOptions;
import com.instana.android.core.InstanaConfig;

public class DotnetNewBinding
{
    public static String getString(String myString)
    {
        return myString + " from java!123";
    }

    public static Boolean setup(Activity activity, String key, String url){
        InstanaConfig config = new InstanaConfig(key,url);
        config.setEnableCrashReporting(true);
        HybridAgentOptions hybridAgentOptions= new HybridAgentOptions("maui","0.0.1");
        if(activity.getApplication() != null){
            Instana.setupInternal(activity.getApplication(),config,hybridAgentOptions);
        }else {
            Log.e("Instana- ","No Application Context available");
        }
        return true;
    }

    public static void setView(String view){
        Instana.setView(view);
    }

    public static void setUserId(String userId){
        Instana.setUserId(userId);
    }

    public static void setUserName(String userName){
        Instana.setUserName(userName);
    }

    public static void setUserEmail(String userEmail){
        Instana.setUserEmail(userEmail);
    }

    public static void setScreenNameAutoCapture(Boolean autoCapture){
        Instana.setAutoCaptureScreenNameEnabled(autoCapture);
    }

    public static void setCrashReporting(Boolean crashReporting){
        Instana.setCrashReportingEnabled(crashReporting);
    }

    public static void setCustomEvent(String eventName){
        CustomEvent event = new  CustomEvent(
                eventName
        );
        Instana.reportEvent(event);
    }
}

File: dotnet-maui-instana-integration/macios/native/NewBinding/NewBinding/DotnetNewBinding.swift
//
//  DotnetNewBinding.swift
//  NewBinding
//
//  Created by .NET MAUI team on 6/18/24.
//

import Foundation
import InstanaAgent

@objc(DotnetNewBinding)
public class DotnetNewBinding : NSObject
{

    @objc
    public static func getString(myString: String) -> String {
        return myString  + " from swift!"
    }
    
    @objc
    public static func setup(key: String,url: String) {
        Instana.setupInternal(key: key, reportingURL: URL(string: url)!, options: InstanaSetupOptions(), hybridOptions: HybridAgentOptions(id: "maui", version: "0.0.1"))
    }
    
    @objc
    public static func setView(view: String) {
        Instana.setView(name: view)
    }

    @objc
    public static func setUserId(userId: String) {
        Instana.setUser(id: userId)
    }
    
    @objc
    public static func setUserName(userName: String) {
        Instana.setUser(name: userName)
    }
    
    @objc
    public static func setUserEmail(userEmail: String) {
        Instana.setUser(email: userEmail)
    }
    
    @objc
    public static func setScreenNameAutoCapture(autoCapture: Bool) {
        //Impl
    }
    
    @objc
    public static func setCrashReporting(crashReporting: Bool) {
        //Impl
    }
    
    @objc
    public static func setCustomEvent(eventName: String) {
        Instana.reportEvent(name: eventName)
    }
}
